src/cdomains/apron/octApronDomain.apron.ml

```diff
@@ -593,4 +562,287 @@ struct
+
+  let forget_vars d vs =
+    (* TODO: forget_all which takes Var arguments instead *)
+    forget_all d (List.map Var.to_string vs)
+end
+
+
+(* Copy-paste from BaseDomain... *)
+type 'a octaproncomponents_t = {
+  oct: D2.t;
+  priv: 'a;
+} [@@deriving eq, ord, to_yojson]
+
+module OctApronComponents (PrivD: Lattice.S):
+sig
+  include Lattice.S with type t = PrivD.t octaproncomponents_t
+  val op_scheme: (D2.t -> D2.t -> D2.t) -> (PrivD.t -> PrivD.t -> PrivD.t) -> t -> t -> t
+end =
+struct
+  type t = PrivD.t octaproncomponents_t [@@deriving eq, ord, to_yojson]
+
+  include Printable.Std
+  open Pretty
+  let hash r  = D2.hash r.oct + PrivD.hash r.priv * 33
+
+
+  let show r =
+    let first  = D2.show r.oct in
+    let third  = PrivD.show r.priv in
+    "(" ^ first ^ ", " ^ third  ^ ")"
+
+  let pretty () r =
+    text "(" ++
+    D2.pretty () r.oct
+    ++ text ", " ++
+    PrivD.pretty () r.priv
+    ++ text ")"
+
+  let printXml f r =
+    BatPrintf.fprintf f "<value>\n<map>\n<key>\n%s\n</key>\n%a<key>\n%s\n</key>\n%a</map>\n</value>\n" (Goblintutil.escape (D2.name ())) D2.printXml r.oct (Goblintutil.escape (PrivD.name ())) PrivD.printXml r.priv
+
+  let name () = D2.name () ^ " * " ^ PrivD.name ()
+
+  let invariant c {oct; priv} =
+    Invariant.(D2.invariant c oct && PrivD.invariant c priv)
+
+  let of_tuple(oct, priv):t = {oct; priv}
+  let to_tuple r = (r.oct, r.priv)
+
+  let arbitrary () =
+    let tr = QCheck.pair (D2.arbitrary ()) (PrivD.arbitrary ()) in
+    QCheck.map ~rev:to_tuple of_tuple tr
+
+  let bot () = { oct = D2.bot (); priv = PrivD.bot ()}
+  let is_bot {oct; priv} = D2.is_bot oct && PrivD.is_bot priv
+  let top () = {oct = D2.top (); priv = PrivD.bot ()}
+  let is_top {oct; priv} = D2.is_top oct && PrivD.is_top priv
+
+  let leq {oct=x1; priv=x3 } {oct=y1; priv=y3} =
+    D2.leq x1 y1 && PrivD.leq x3 y3
+
+  let pretty_diff () (({oct=x1; priv=x3}:t),({oct=y1; priv=y3}:t)): Pretty.doc =
+    if not (D2.leq x1 y1) then
+      D2.pretty_diff () (x1,y1)
+    else
+      PrivD.pretty_diff () (x3,y3)
+
+  let op_scheme op1 op3 {oct=x1; priv=x3} {oct=y1; priv=y3}: t =
+    {oct = op1 x1 y1; priv = op3 x3 y3 }
+  let join = op_scheme D2.join PrivD.join
+  let meet = op_scheme D2.meet PrivD.meet
+  let widen = op_scheme D2.widen PrivD.widen
+  let narrow = op_scheme D2.narrow PrivD.narrow
+end
+
+
+module Var =
+struct
+  include Var
+
+  let equal x y = Var.compare x y = 0
+end
+
+module type VarMetadata =
+sig
+  type t
+  val var_name: t -> string
+end
+
+module VarMetadataTbl (VM: VarMetadata) =
+struct
+  module VH = Hashtbl.Make (Var)
+
+  let vh = VH.create 113
+
+  let make_var metadata =
+    let var = Var.of_string (VM.var_name metadata) in
+    VH.replace vh var metadata;
+    var
+
+  let find_metadata var =
+    VH.find_option vh var
+end
+
+module GVM =
+struct
+  include CilType.Varinfo
+  let var_name g = g.vname
+end
+
+module GV =
+struct
+  include VarMetadataTbl (GVM)
+
+  let make g = make_var g
+end
```

Comment:
I think you cherry-picked too much stuff here, this isn't necessary here, is it?